base/compiler/abstractinterpretation.jl

```diff
@@ -869,6 +869,15 @@ function bail_out_const_call(interp::AbstractInterpreter, result::MethodCallResu
         if isa(result.rt, Const) || call_result_unused(si)
             return true
         end
+    elseif result.rt === Bottom
+        if is_terminates(result.effects) && is_effect_free(result.effects)
```

Comment:
```suggestion
        if is_foldable(result.effects)
```
? Since we can expect `result.effects.consistent === ALWAYS_TRUE` when `result.rt === Bottom`.

Reply:
Hm, well, this might be fine as is, since the whole point of this change is to prevent concrete-evaluation rather than enabling it.

Reply:
Yeah, consistent should be implied by the rt condition, but it's not mandatory, so I just figured it was better not to check for it.